far/vmenu.cpp

```diff
@@ -3048,4 +3129,72 @@ TEST_CASE("find.nearest.selectable.item")
 	}
 }
 
+TEST_CASE("intersect.segments")
+{
+	static constexpr struct test_data
+	{
+		std::pair<int, int> A, B, Intersection;
+	} TestDataPoints[] =
+	{
+		{ { 10, 20 }, { -1, 5 }, { 0, 0 } },
+		{ { 10, 20 }, { -1, 10 }, { 0, 0 } },
+		{ { 10, 20 }, { -1, 15 }, { 10, 15 } },
+		{ { 10, 20 }, { -1, 20 }, { 10, 20 } },
+		{ { 10, 20 }, { -1, 25 }, { 10, 20 } },
+		{ { 10, 20 }, { 10, 15 }, { 10, 15 } },
+		{ { 10, 20 }, { 10, 20 }, { 10, 20 } },
+		{ { 10, 20 }, { 10, 25 }, { 10, 20 } },
+		{ { 10, 20 }, { 15, 20 }, { 15, 20 } },
+		{ { 10, 20 }, { 15, 25 }, { 15, 20 } },
+		{ { 10, 20 }, { 20, 25 }, { 0, 0 } },
+		{ { 10, 20 }, { 25, 30 }, { 0, 0 } },
+	};
+
+	for (const auto& TestDataPoint : TestDataPoints)
+	{
+		REQUIRE(TestDataPoint.Intersection == Intersect(TestDataPoint.A, TestDataPoint.B));
+		REQUIRE(TestDataPoint.Intersection == Intersect(TestDataPoint.B, TestDataPoint.A));
+	}
+}
+
+TEST_CASE("markup.slice.boundaries")
+{
+	static struct test_data
+	{
+		std::pair<int, int> Segment;
+		std::vector<std::pair<int, int>> Slices;
```

Comment:
Do we need vectors here?
The data is static, so spans, initializer_lists or arrays should probably do.

Reply:
Ah, `initializer_list` works with `const`, but not with `constexpr` here. I originally played with `constexpr`, could not make it work, and gave up. Went with `vector`s and lost `const`. :( Thank you for calling this out. fixed.